sources/Context.cpp

```diff
@@ -17,11 +17,26 @@
 
 #include "Context.h"
 
+#include "Log.h"
+
 #include <tuple>
 
 namespace dfl {
-Context::Context(const std::string& networkFilepath) : networkManager_(networkFilepath) {
-  networkManager_.onNode(std::bind(&Context::processSlackNode, this, std::placeholders::_1));
+Context::Context(const std::string& networkFilepath, const std::string& configFilepath) :
+    networkManager_(networkFilepath),
+    config_(configFilepath),
+    slack_node_{} {
+  auto found_slack_node = networkManager_.getSlackNode();
+  if (found_slack_node.is_initialized() && !config_.isAutomaticSlackBusOn()) {
+    slack_node_ = *found_slack_node;
+  } else {
+    // slack node not given in iidm or not requested: we must compute it ourselves
+    if (!found_slack_node.is_initialized() && !config_.isAutomaticSlackBusOn()) {
+      // TODO(lecourtoisflo) use warning without restriction for language
+      LOG(warn) << "Automatic slack node requested but not present in network input file" << LOG_ENDL;
+    }
```

Comment:
The case if (found_slack_node.is_initialized() && config_.isAutomaticSlackBusOn()) is missing I think

Reply:
Done next line